framework/src/base/FEProblem.C

```diff
@@ -240,6 +240,8 @@ FEProblem::~FEProblem()
        it != _random_data_objects.end(); ++it)
     delete it->second;
 
+  if (_xfem != NULL)
+    delete _xfem;
```

Comment:
How about we use a smart pointer here. FEProblem deletes an object it did not allocate itself. This is just asking for trouble.